streamz/core.py

```diff
@@ -1231,6 +1231,111 @@ def cb(self):
             yield self._emit(x)
 
 
+@Stream.register_api()
+class to_kafka(Stream):
+    """ Writes data in the stream to Kafka
+
+    This stream accepts a string or bytes object. Call ``flush`` to ensure all
+    messages are pushed. Responses from Kafka are pushed downstream.
+
+    Parameters
+    ----------
+    topic : string
+        The topic which to write
+    producer_config : dict
+        Settings to set up the stream, see
+        https://docs.confluent.io/current/clients/confluent-kafka-python/#configuration
+        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        Examples:
+        bootstrap.servers: Connection string (host:port) to Kafka
+
+    Examples
+    --------
+    >>> from streamz import Stream
+    >>> ARGS = {'bootstrap.servers': 'localhost:9092'}
+    >>> source = Stream()
+    >>> kafka = source.map(lambda x: str(x)).to_kafka('test', ARGS)
+    <to_kafka>
+    >>> for i in range(10):
+    ...     source.emit(i)
+    >>> kafka.flush()
+    """
+    Result = namedtuple('Result', ['err', 'msg'])
+
+    def __init__(self, upstream, topic, producer_config, **kwargs):
+        import confluent_kafka as ck
+
+        self.topic = topic
+        self.producer = ck.Producer(producer_config)
+
+        Stream.__init__(self, upstream, ensure_io_loop=True, **kwargs)
+        self.stopped = False
+        self.polltime = 0.5
+        self.loop.add_callback(self.poll)
+
+    @gen.coroutine
+    def poll(self):
+        while True:
+            # executes callbacks for any delivered data, in this thread
+            # if no meesages were sent, nothing happens
+            self.producer.poll(0)
+            if self.stopped:
+                # kill producer?
+                break
+            yield gen.sleep(self.polltime)
+
+    def update(self, x, who=None):
+        future = gen.Future()
+
+        def _():
+            while True:
```

Comment:
I'm curious, why the `while True` here?  

If there is a `BufferError` or another exception we want to try this again and again?

Reply:
Only for BufferError, I think: the producer queue is full, and so message can't be sent at the moment. This implies that the function should be a coroutine and gen-sleep for a while, fixed in the coming commit.

Reply:
Ah, that makes sense.